lib/krikri/mapper.rb

```diff
@@ -62,6 +63,37 @@ def map(name, records)
     ##
     # An application-wide registry of defined mappings
     Registry = Class.new(Krikri::Registry)
-    
+
+    ##
+    # A SoftwareAgent to run mapping processes.
+    #
+    # @see: Krikri::SoftwareAgent, Krikri::Activity
+    class Agent
+      include SoftwareAgent
+
+      attr_reader :name, :generator_uri
+
+      def initialize(opts = {})
+        @name = opts.fetch(:name).to_sym
+        @generator_uri = RDF::URI(opts.fetch(:generator_uri))
+      end
+
+      def run(activity_uri = nil)
+        Krikri::Mapper.map(name, records).each do |rec|
+          rec.mint_id! if rec.node?
+          rec << RDF::Statement(rec, RDF::PROV.wasGeneratedBy, activity_uri) if
+            activity_uri
+          rec.save
+        end
+      end
+
+      def records
```

Comment:
Looking at your concern raised in the specs for this and `Krikri::ProvenanceQueryClient`, maybe  we should add a method that returns a new `Krikri::ProvenanceQueryClient` instance. Then you can ensure that it responds to `execute` as a subclass of `SPARQL::Client` in its own spec, maybe like [here](https://github.com/ruby-rdf/sparql-client/blob/develop/spec/client_spec.rb#L179-L188).